### PR TITLE
all 3 assignments done and ready to be merged upon approval

### PR DESCRIPTION
**Pull request by JBrady**

### CODE REVIEW

File: code-challenges.js [hunk 1]
```diff
@@ -9,9 +9,46 @@
...
+
+// if (temperature.above === 350) {
+//     console.log(`${temperature.above} is above boiling point`)
+// } else if (temperature.below === 35) {
+//     console.log(`${temperature.below} is below boiling point`)
+// } else {
+//     console.log(`${temperature.boiling} is boiling point`)
+// }
+
+// console.log(message)
```
Comment (WalkerTexas-coder) [inline code-challenges.js:49]: Cool stuff, one reason this isn't working is because message is defined three times.

File: code-challenges.js [hunk 2]
```diff
@@ -9,9 +9,46 @@
...
+
+if (temperature == 212) {
+    console.log("212 is boiling point")
+} else if (temperature == 350) {
+    console.log("350 is above boiling point")
+} else if (temperature == 35) {
+    console.log("35 is below boiling point")
+} else {
+    console.log("Those are your only choices")
+}
```
Comment (WalkerTexas-coder) [inline code-challenges.js:27]: I like this, but what would make this more dynamic is if you called on the >, < operators. 
```javascript
if (temperature == 212) {
    console.log("input is boiling point")
} else if (temperature > 212) {
    console.log("input is above boiling point")
} else if (temperature < 212) {
    console.log("input is below boiling point")
} else {
    console.log("Those are your only choices")
}
```

File: code-challenges.js [hunk 3]
```diff
@@ -46,3 +106,12 @@ var numberRefactor2 = "hello"
...
 // var numberRefactor2 = 24
+
+// This is similar to what I have above. I'm using an if statement to find out it number1 is greater than number2. If it is, then I'm going to console.log the result of number1 minus number2. I'm adding else-if to the evaluation to check if number1 is less than number2. If it's less, then I'm going to console.log the result of number2 minus number1. If neither of those can be true (because there's a string instead of a number) then I'll print out "Your input is not a number".
+if (numberRefactor1 > numberRefactor2) {
+    console.log(numberRefactor1 - numberRefactor2)
+} else if (numberRefactor1 < numberRefactor2) {
+    console.log(numberRefactor2 - numberRefactor1)
+} else {
+    console.log("Your input is not a number")
+}
```
Comment (WalkerTexas-coder) [inline code-challenges.js:117]: Great work!

File: coding-concepts.js [hunk 4]
```diff
@@ -8,44 +8,45 @@
...
 // --------------------2) What will this log?
 
 var greeting = "Hello World!"
-// console.log(greeting[3])
+console.log(greeting[3])
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer: This should print out "Hello World" 3 times in a row
+// b) Verify and explain: I was wrong. It's actually telling console.log to return the character located in the index of 3, which is the letter 'l'.
```
Comment (WalkerTexas-coder) [inline coding-concepts.js:23]: Good work!

File: coding-concepts.js [hunk 5]
```diff
@@ -8,44 +8,45 @@
...
 // --------------------4) What will this log?
 
 var weekendDays = ["saturday", "sunday"]
 // console.log(weekendDays.toUpperCase())
+console.log(weekendDays.join(", ").toUpperCase())
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer: This will return the entire array converted to uppercase letters.
+// b) Verify and explain: This returns the TypeError "weekendDays.toUpperCase is not a function". That's because the .toUpperCase() method is suppossed to be used on strings, not arrays. If you use .join(", ") first, then use .toUpperCase() it returns the output I described in my answer above.
```
Comment (WalkerTexas-coder) [inline coding-concepts.js:43]: Excellent!

File: interview-questions.md [hunk 6]
```diff
@@ -7,60 +7,60 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is git? What is the difference between git and Github?
 
-  Your answer:
+  Your answer: Git is a version control system created by Linus Torvalds to assist in the development of the Linux OS. It's taken over as the industry standard from an older system called Subversion. Git's main advantage is that it's decentralized, meaning it doesn't need a single server for people to work off of. It's decentralized nature is why GitHub exists. Anyone can push thier changes up to GitHub and allow anyone else to contribute to thier projects. GitHub is the cloud service that runs Git.
 
-  Researched answer:
+  Researched answer: Basically the same thing I wrote above. Git is the software and GitHub is the cloud service.
```
Comment (WalkerTexas-coder) [inline interview-questions.md:12]: I can appreciate the good work you did. Your researched answer is an opportunity to dive deeper on the subject or even tangential subjects. How does the cloud service work? Who made it?  Is it related to git? etc...

File: interview-questions.md [hunk 7]
```diff
@@ -7,60 +7,60 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 
 
 4. What is a built-in method?
 
-  Your answer:
+  Your answer: A piece of code that performs a specific action that's built into a programming language to make it easier for the user.
 
-  Researched answer:
+  Researched answer: Javascript built-in methods are actions that can be performed by attaching them to "objects" (we havent covered objects yet, but I don't know how else to phrase it). One helpful sentence I read stated "A property is what an object has, while a method is what an object does". This helped clear up some confusion around '.length' and why it's referred to in the sylabus as a 'property' and not a 'method'.
```
Comment (WalkerTexas-coder) [inline interview-questions.md:36]: Awesome deep dive!

**Review by WalkerTexas-coder — APPROVED** [review]
John, well done! I really appreciate your creative exploration. Even more I appreciate your communication on the different things you tried. I left a number of specific code notes below for your reference and for encouragement. You've done a great job here keep up the good work! Your PR is approved

File: interview-questions.md (hunk 6 above)
Comment (JBrady) [reply interview-questions.md:12]: Got it. I'll expand on it more thoroughly next assignment. Thx!

**Pull request merged**